src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -0,0 +1,570 @@
+            "ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedCluster = self.cmd.get_models(
+            "ManagedCluster",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        # pylint: disable=line-too-long
+        self.Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties = self.cmd.get_models(
+            "Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ExtendedLocation = self.cmd.get_models(
+            "ExtendedLocation",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ExtendedLocationTypes = self.cmd.get_models(
+            "ExtendedLocationTypes",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+
+
+# pylint: disable=too-few-public-methods
+class AKSCreateParameters:
+    # Used to store original function parameters, in the form of attributes of this class, which can be
+    # obtained through a.xxx (a is an instance of this class, xxx is the original parameter name).
+    # Note: The attributes of this class should not be modified once they are initialized.
+    def __init__(self, data: dict):
+        for name, value in data.items():
+            setattr(self, name, value)
+
+
+class AKSCreateContext:
+    # Used to store dynamically inferred/completed parameters (i.e. not specified by the user), intermediate
+    # variables and a copy of the original function parameters.
+    # To dynamically infer/complete a parameter or check the validity of a parameter, please provide a function
+    # named `get_xxx`, where `xxx` is the parameter name.
+    # Attention: When checking the validity of parameters in the `get_xxx` function, please use the `get_param`
+    # function to obtain the values of other parameters to be checked to avoid circular calls.
+    # Note: The update of parameters and intermediate variables in the command implementation should be achieved
+    # by operating the instance of this class.
+    def __init__(self, cmd: AzCliCommand, raw_parameters: dict):
+        self.cmd = cmd
+        self.raw_param = AKSCreateParameters(raw_parameters)
+        self.intermediates = dict()
+        self.mc = None
+
+    def attach_mc(self, mc):
+        self.mc = mc
+
+    def get_intermediate(self, variable_name: str, default_value: Any = None):
+        if variable_name not in self.intermediates:
+            msg = "The intermediate '{}' does not exist! Return default value '{}'!".format(
```

Comment:
remove the !

Reply:
Sure, fixed in commit [4cf8354](https://github.com/Azure/azure-cli/pull/19259/commits/4cf8354c083f122894111a25d737eeaabb3df5d2).